### PR TITLE
Fix nonlocal incompatibility

### LINKED ISSUE #2189: nonlocal incompatibility
###  Here is the list of incompatibility of `nonlocal` between RustPython and CPython 

```python
  def f(a):
    nonlocal a
```
* CPython 3.8
  `SyntaxError: name 'a' is parameter and nonlocal`
* RustPython 
`SyntaxError: no binding for nonlocal 'a' found at line 0 column 0`

```python
def f():
  a = 3
  def g():
    print(a)
    nonlocal a
  g()
```
* CPython 3.8
`SyntaxError: name 'a' is used prior to nonlocal declaration`
* RustPython
 `No Error. f() prints 3`

```python
  def f():
    a: int
    nonlocal a
```
* CPython 3.8
`SyntaxError: annotated name 'a' can't be nonlocal`

* RustPython
`SyntaxError: no binding for nonlocal 'a' found at line 0 column 0`

```python
  def f():
    a = 3
    def g():
      a = 4
      nonlocal a
    g()
    print(a)
```
* CPython 3.8
`SyntaxError: name 'a' is assigned to before nonlocal declaration`
* RustPython
 `No error. f() prints 4`

### PR DESCRIPTION
**Pull request by minoring**
Fix #2189 and #2183

### CODE REVIEW

File: compiler/src/symboltable.rs [hunk 1]
```diff
@@ -954,10 +954,10 @@ impl SymbolTableBuilder {
         // Some checks:
         let containing = table.symbols.contains_key(name);
         if containing {
```
Comment (coolreader18) [inline compiler/src/symboltable.rs:956]: I think you can replace this with `if let Some(symbol) = table.symbols.get(name) {`

**Review by coolreader18 — APPROVED** [review]
Looks good!

File: compiler/src/symboltable.rs (hunk 1 above)
Comment (minoring) [reply compiler/src/symboltable.rs]: Thank you for your tip :) 
I made refactoring to replace `containing` with `Some(symbol)`
PTAL!

**Comment by coolreader18** [comment]
Nice work :)

**Pull request merged**